app/controllers/employments/UpdateEmploymentController.scala

```diff
@@ -119,14 +121,16 @@ class UpdateEmploymentController @Inject()(employmentService: EmploymentService,
   def addTelephoneNumber(): Action[AnyContent] = (authenticate andThen validatePerson).async {
     implicit request =>
       for {
-        employmentId <- journeyCacheService.mandatoryValueAsInt(EndEmployment_EmploymentIdKey)
+        employmentId <- journeyCacheService.mandatoryJourneyValueAsInt(EndEmployment_EmploymentIdKey)
         telephoneCache <- journeyCacheService.optionalValues(UpdateEmployment_TelephoneQuestionKey, UpdateEmployment_TelephoneNumberKey)
       } yield {
         implicit val user = request.taiUser
-        Ok(views.html.can_we_contact_by_phone(Some(user), telephoneNumberViewModel(employmentId),
-          YesNoTextEntryForm.form().fill(YesNoTextEntryForm(telephoneCache.head, telephoneCache(1)))))
+        employmentId match {
```

Comment:
we could use a fold here, here begins the debate on which is more readable?

Reply:
I think the Case statement is more readable though fold is compact.